custom-plugins/cms131v14_diabetes_eye_exam/protocols/cms131v14_protocol.py

```diff
@@ -0,0 +1,690 @@
+        )
+
+        return card.apply()
+
+    def _patient_has_any_code_in_value_sets(
+        self, patient: Patient, value_set_classes: list
+    ) -> bool:
+        """
+        Helper to check if patient has any active condition coding present in given value sets.
+        
+        Args:
+            patient: The patient to check
+            value_set_classes: List of value set classes (e.g., [PalliativeOrHospiceCare, HospiceCareAmbulatory])
+        
+        Returns:
+            True if any active condition coding matches any of the value sets
+        """
+        try:
+            conditions = Condition.objects.for_patient(patient.id).active()
+
+            for condition in conditions:
+                for coding in condition.codings.all():
+                    for value_set_class in value_set_classes:
+                        if self._coding_in_value_set(coding, value_set_class):
+                            log.info(
+                                f"CMS131v14: Found matching code {coding.code} in value set for patient {patient.id}"
+                            )
+                            return True
+
+            return False
+        except Exception as e:
+            log.error(f"CMS131v14: Error checking codes in value sets: {str(e)}")
+            return False
+
+    def _coding_in_value_set(self, coding: ConditionCoding, value_set_class) -> bool:
+        """
+        Helper to check if a single coding belongs to a value set.
+        
+        Args:
+            coding: The ConditionCoding to check
+            value_set_class: Value set class to check against
+        
+        Returns:
+            True if the coding is found in the value set
+        """
+        try:
+            # Normalize the coding system for comparison
+            normalized_system = coding.system.replace("-", "").upper()
+
+            # Check common coding systems
+            if normalized_system in ["ICD10", "ICD10CM"]:
+                codes = getattr(value_set_class, "ICD10CM", set())
+                normalized_code = coding.code.replace(".", "")
+                codes_normalized = {code.replace(".", "") for code in codes}
+                return normalized_code in codes_normalized
+            elif normalized_system in ["SNOMEDCT", "SNOMED"]:
+                codes = getattr(value_set_class, "SNOMEDCT", set())
+                return coding.code in codes
+            elif normalized_system == "CPT":
+                codes = getattr(value_set_class, "CPT", set())
+                return coding.code in codes
+            elif normalized_system == "HCPCS":
+                codes = getattr(value_set_class, "HCPCS", set())
+                return coding.code in codes
+
+            return False
+        except Exception as e:
+            log.error(f"CMS131v14: Error checking coding in value set: {str(e)}")
+            return False
+
+    def _calculate_age(self, patient: Patient) -> int | None:
```

Comment:
The patient model has an [`age_at()` method](https://github.com/canvas-medical/canvas-plugins/blob/6cd6aefe4b6f9a359c5e3a0e7387b72c2f475648/canvas_sdk/v1/data/patient.py#L119-L134) that can be used, avoiding the need for this method.